lib/tasks/subscriptions.rake

```diff
@@ -7,4 +7,32 @@ namespace :subscriptions do
       subscription.update!(unique_id: subscription.customer.customer_id)
     end
   end
+
+  # NOTE: Ability to create invoices in the future.
+  # ie: bundle exec rake "subscriptions:generate_invoice[timestamp, external_id1, external_id2, ...]"
+  desc 'Generate invoice for a specific timestamp'
+  task :generate_invoice, [:timestamp] => :environment do |_task, args|
+    abort "Missing timestamp and external subscription ids\n\n" unless args[:timestamp]
+    abort "Missing external subscription ids\n\n" if args.extras.blank?
+
+    subscriptions = Subscription.where(external_id: args.extras)
+
+    abort "External subscription ids not found\n\n" if subscriptions.blank?
+    abort "Subscriptions don't belong to the same customer\n\n" if subscriptions.pluck(:customer_id).uniq.count > 1
+
+    result = Invoices::SubscriptionService.new(subscriptions:, timestamp: args[:timestamp].to_i, recurring: false).create
```

Comment:
Layout/LineLength: Line is too long. [121/120] (https://rubystyle.guide#max-line-length)